test/fw/ptl/utils/plugins/ptl_test_loader.py

```diff
@@ -190,7 +196,19 @@ def prepareTestLoader(self, loader):
         old_loadTestsFromNames = loader.loadTestsFromNames
 
         def check_loadTestsFromNames(names, module=None):
-            rv = old_loadTestsFromNames(names, module)
+            tests_dir = names
+            if not self.testfiles:
+                ptl_test_dir = __file__
+                ptl_test_dir = os.path.join(ptl_test_dir.split('ptl')[0],
+                                            "ptl", "tests")
+                user_test_dir = os.environ.get("PTL_TESTS_DIR", None)
+                if user_test_dir and os.path.isdir(user_test_dir):
+                    tests_dir += [user_test_dir]
+                if os.path.isdir(ptl_test_dir):
+                    tests_dir += [ptl_test_dir]
```

Comment:
So, this will mean that if PTL_TESTS_DIR is set, then tests will be picked from \<ptl install\>/tests as well as PTL_TESTS_DIR. Is that really what we want? I kind of feel that if users specify PTL_TESTS_DIR, then that should be the ONLY directory where tests should be searched for. What do you think?

Reply:
We are giving 3 dir to the loader (in a specific order: 1. current dir 2. PTL_TESTS_DIR 3. PTL install dir) to find given tests and loader will load tests from only first dir it find test rest will be ignored for perticular test.

Reply:
Ok, but I feel that it's more intuitive to only look at PTL_TESTS_DIR if it is set by the user. If unset, we can look in current dir + the default install location.

Reply:
Ok, but what if the given test is not in PTL_TESTS_DIR but in default install location?

Reply:
then we'll say test not found, and that'll be expected behavior. If you disagree then we can ask for an opinion on forums.

Reply:
@agrawalravi90 that's what we discussed on forums if I recall it correctly and it concluded that we will give all 3 directly to the loader and let loader find test from one of them and if not found in all 3 then only give test not found error. But saying that I am happy re-discuss that but current code are in line with the current design so if we want to change what you say then first we need to change the design.

Reply:
if this has already been discussed then never-mind, I don't want to delay the PR. Let's check it in as is and if somebody else feels that it should change or gets confused by the behavior then we can bring this up again.